src/shapes/active_selection.class.js

```diff
@@ -52,13 +52,49 @@
       this.setCoords();
     },
 
+    /**
+     * @private
+     * @param {string} key
+     * @param {*} value
+     */
+    _set: function (key, value) {
+      var prev = this[key];
+      this.callSuper('_set', key, value);
+      if (prev !== this[key] && this.canvas && this.canvas.preserveObjectStacking
+        && this.stateProperties.indexOf(key) > -1) {
+        var invalidationContext = {
+          target: this,
+          key: key,
+          value: value,
+          prevValue: prev
+        };
+        var invalidatedGroups = [];
+        this.forEachObject(function (object) {
+          var group = object.__owningGroup;
+          if (group && invalidatedGroups.indexOf(group) === -1) {
+            group.invalidate(invalidationContext);
```

Comment:
maybe I want to add a check for objects that are group de-facto but aren't really groups:
```suggestion
            group.invalidate && group.invalidate(invalidationContext);
```